src/main/python/ayab/engine/pattern.py

```diff
@@ -149,44 +152,44 @@ def num_colors(self, num_colors):
             self.__update_pattern_data()
 
     @property
-    def alignment(self):
+    def alignment(self)->Alignment:
         return self.__alignment
 
     @alignment.setter
-    def alignment(self, alignment):
+    def alignment(self, alignment:Alignment)->None:
         """
         set the position of the pattern
         """
         self.__alignment = alignment
         self.__update_pattern_data()
 
     @property
-    def pat_start_needle(self):
+    def pat_start_needle(self)->int:
         return self.__pat_start_needle
 
     @property
-    def pat_end_needle(self):
+    def pat_end_needle(self)->int:
         return self.__pat_end_needle
 
     @property
-    def knit_start_needle(self):
+    def knit_start_needle(self)->int:
         return self.__knit_start_needle
 
     @property
-    def knit_end_needle(self):
+    def knit_end_needle(self)->int:
         return self.__knit_end_needle
 
     @property
-    def pat_height(self):
+    def pat_height(self)->int:
         return self.__pat_height
 
     @property
-    def pat_width(self):
+    def pat_width(self)->int:
         return self.__pat_width
 
     @property
-    def pattern_expanded(self):
+    def pattern_expanded(self)->list[bitarray]:
         return self.__pattern_expanded
 
-    def array2rgb(self, a):
+    def array2rgb(self, a:list[int])->int:
```

Comment:
technically `a` is a `npt.NDArray[int_]` but the type resolution fell apart when I tried that. Also, you could totally pass any ol' list to this, so I'm ok with it being "wrong"